internal/main.go

```diff
@@ -76,6 +77,15 @@ func main() {
 		}
 	}
 
+	azureProvisioner := azure.NewAzureProvisioner(&logger)
+
+	if flags.oem.String() == "azure" && flags.stage == "disks" {
```

Comment:
It's not ideal to have Azure-specific code in `main`.  It'd be nice to abstract this out so other platforms could have their own hooks; see https://github.com/coreos/bugs/issues/2130 and point 2 of https://github.com/coreos/bugs/issues/2234.